### PR TITLE
Add a version check to provide good errors on version incompatibility

### PR DESCRIPTION
**Pull request by RussTheAerialist**
partial fix for #58

### CODE REVIEW

**Comment by benlower** [comment]
what else needs to be done to make this a 'full' fix?

**Comment by RussTheAerialist** [comment]
I don't quite know yet. Waiting on the non-8.0 ones that are failing

**Comment by t8y8** [comment]
Working on some stats to veirfy

**Comment by t8y8** [comment]
Counter({'8.0': 46, '8.2': 2, '8.1': 1, '9.0': 1})

One of the workbooks was 9.0, I'll send that one to @RussTheAerialist to see if it's related or not

**Comment by RussTheAerialist** [comment]
Thanks @t8y8 !  I'll either get a fix done this weekend or Monday (depending on how my weekend goes)

File: tableaudocumentapi/xml_open.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import xml.etree.ElementTree as ET
+import zipfile
+
+from tableaudocumentapi import xfile
+
+MIN_SUPPORTED_VERSION = Version("9.0")
+
+
+class VersionNotSupportedException(Exception):
```
Comment (t8y8) [inline tableaudocumentapi/xml_open.py]: TableauFileVersionNotSupported or TableauVersionNotSupported?

File: tableaudocumentapi/xml_open.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    pass
+
+
+def xml_open(filename):
```
Comment (t8y8) [inline tableaudocumentapi/xml_open.py]: I think this should just go in the xfile module as the functionality is all around opening around zip packaging.

(And that's a great cleanup I didn't get to, thank you!)

File: tableaudocumentapi/xml_open.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+try:
+    from distutils2.version import NormalizedVersion as Version
```
Comment (t8y8) [inline tableaudocumentapi/xml_open.py]: This feels heavy to me for what we're doing.

The strings are ordered and can be compared ("8.2" < "9.0 == True), can we just do
```
MIN_SUPPORTED_VERSION = "9.0"
...
 file_version = tree.getroot().attrib.get('version', '0.0')
    print(file_version, MIN_SUPPORTED_VERSION) # And remove this print, it's just debugging?
    if file_version < MIN_SUPPORTED_VERSION:
        raise VersionNotSupportedException(file_version)
    return tree
```

File: tableaudocumentapi/xml_open.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+MIN_SUPPORTED_VERSION = Version("9.0")
+
+
+class VersionNotSupportedException(Exception):
+    pass
+
+
+def xml_open(filename):
+    # Determine if this is a twb or twbx and get the xml root
```
Comment (t8y8) [inline tableaudocumentapi/xml_open.py]: nit: tds and tdsx too... maybe just say "if this is an xfile or not" ;) -- we really need a word for those 'x' files we have

File: tableaudocumentapi/xml_open.py (hunk 3 above)
Comment (RussTheAerialist) [inline tableaudocumentapi/xml_open.py]: If only t'were so simple. (sorry, I just watched _Hail, Caeser_)

```python
'8.0' < '10.0' == False
```

File: tableaudocumentapi/xml_open.py (hunk 4 above)
Comment (RussTheAerialist) [inline tableaudocumentapi/xml_open.py]: oh, doh, I forgot to update the comment when I moved the code over

File: tableaudocumentapi/xml_open.py (hunk 2 above)
Comment (RussTheAerialist) [inline tableaudocumentapi/xml_open.py]: fair point, I'll move it

File: tableaudocumentapi/xml_open.py (hunk 3 above)
Comment (t8y8) [inline tableaudocumentapi/xml_open.py]: Argh. Alrighty, leverage the stdlib awayyyyyy

**Comment by RussTheAerialist** [comment]
Btw, the last commit makes this a complete fix for #58

File: tableaudocumentapi/xfile.py [hunk 5]
```diff
@@ -16,10 +38,10 @@ def temporary_directory(*args, **kwargs):
         shutil.rmtree(d)
 
 
-def find_file_in_zip(zip):
-    for filename in zip.namelist():
+def find_file_in_zip(zip_file):
```
Comment (RussTheAerialist) [inline tableaudocumentapi/xfile.py]: This change is because `zip` is a built-in so we should not use that as a name of the argument.

**Comment by t8y8** [comment]
:rocket:

File: tableaudocumentapi/xfile.py (hunk 5 above)
Comment (t8y8) [inline tableaudocumentapi/xfile.py]: That's coo' w. me

**Comment by benlower** [comment]
🚀

**Comment by RussTheAerialist** [comment]
This is the last fix before 0.2 is cut (unless someone speaks up).  I'm going to cut the release in the morning.

**Pull request merged**